### ISSUE TITLE
Set orthogonal or perspective views

### ISSUE THREAD

**Issue by cysidron1** [gabotechs/react-stl-viewer#44]
Hello there - in my app I'd like to add a button for users to reset the view to an orthogonal view. In another use case, I'd like to set the orthogonal view when the stl file loads.

For example, in OpenSCAD there is a button that does something like this:

https://github.com/gabotechs/react-stl-viewer/assets/74187869/76aaa8fe-dcc6-4582-89cf-54badf800833

Is this possible currently, or would it require building out more functionality?

**Comment by gabotechs** [comment]
Unfortunately, right now it's not possible, and depending on the approach chosen for implementing this, it might be challenging.

Making the camera position a controlled input would probably have some serious performance impact as the rate at which React will need to update the camera position state would be very high. I think a good approach for solving this problem would be to imperatively set the camera state, something like:

```tsx
function Component() {
  const ref = useRef()

  function setCameraState() {
    ref.current?.setCameraState(...)
  }

  return <StlViewer ref={ref}/>
}
```

How does that approach sound like? I might not be able to tackle this immediately though...

**Comment by cysidron1** [comment]
That implementation looks ideal to me! No worries if you can’t get around
to it immediately. By when do you think you could?

**Comment by gabotechs** [comment]
> That implementation looks ideal to me! No worries if you can’t get around
to it immediately. By when do you think you could?

It's likely that I'll be able to ship this within the next couple months, but can't promise anything...

**Comment by gabotechs** [comment]
I've been able to ship this https://github.com/gabotechs/react-stl-viewer/pull/47 (released under the 2.5.0 tag), there is a small example about how that works in the PR description.

Thanks for suggesting this! it was a good addition.

Closing this as completed, feel free to open another issue if you find some gap in the current implementation.
